src/EnergyPlus/ExtendedHI.cc

```diff
@@ -0,0 +1,538 @@
+                    0.0,
+                    Tc);
+                Rs = (Tc - Ts) / (Q - Qv(Ta, Pa));
+                Real64 Ps = Pc - (Pc - Pa) * Zs(Rs) / (Zs(Rs) + Za_un);
+                if (Ps > phi_salt * pvstar(Ts)) {
+                    General::SolveRoot(
+                        state,
+                        tol,
+                        maxIter,
+                        SolFla,
+                        Ts,
+                        [&](Real64 Ts) { return (Ts - Ta) / Ra_un(Ts, Ta) + (phi_salt * pvstar(Ts) - Pa) / Za_un - (Q - Qv(Ta, Pa)); },
+                        0.0,
+                        Tc);
+                    Rs = (Tc - Ts) / (Q - Qv(Ta, Pa));
+                }
+                return Rs;
+            } else {
+                Rs = 0.0;
+                dTcdt = (1.0 / C) * flux3;
+                return dTcdt;
+            }
+        }
+    }
+
+    // Convert the find_T function
+    Real64 find_T(EnergyPlusData &state, int eqvar_name, Real64 eqvar)
+    {
+        Real64 T;
+        int SolFla;
+
+        if (eqvar_name == static_cast<int>(EqvarName::Phi)) {
+            General::SolveRoot(
+                state, tol, maxIter, SolFla, T, [&](Real64 T) { return find_eqvar_phi(state, T, 1.0) - eqvar; }, 0.0, 240.0);
+        } else if (eqvar_name == static_cast<int>(EqvarName::Rf)) {
+            General::SolveRoot(
+                state,
+                tol,
+                maxIter,
+                SolFla,
+                T,
+                [&](Real64 T) { return (find_eqvar_Rf(state, T, std::min(1.0, Pa0 / pvstar(T)))) - eqvar; },
+                230.0,
+                300.0);
+        } else if (eqvar_name == static_cast<int>(EqvarName::Rs)) {
+            General::SolveRoot(
+                state, tol, maxIter, SolFla, T, [&](Real64 T) { return find_eqvar_rs(state, T, Pa0 / pvstar(T)) - eqvar; }, 295.0, 350.0);
+        } else {
+            General::SolveRoot(
+                state, tol, maxIter, SolFla, T, [&](Real64 T) { return find_eqvar_dTcdt(state, T, Pa0 / pvstar(T)) - eqvar; }, 340.0, 1000.0);
+        }
+
+        return T;
+    }
+
+    Real64 heatindex(EnergyPlusData &state, Real64 Ta, Real64 RH)
+    {
+
+        // Ta: temperature in Kelvin
+        // RH: relative humidity in range of 0.0 to 1.0
+        // The function computes the extended heat index, in Kelvinn
+
+        auto const HVACSystemRootSolverBackup = state.dataRootFinder->HVACSystemRootFinding.HVACSystemRootSolver;
+        state.dataRootFinder->HVACSystemRootFinding.HVACSystemRootSolver = HVACSystemRootSolverAlgorithm::Bisection;
+        int eqvar_name = find_eqvar_name(state, Ta, RH);
+        Real64 eqvar_value = find_eqvar_value(state, Ta, RH);
+
+        Real64 T = find_T(state, eqvar_name, eqvar_value);
+
+        if (Ta == 0.0) T = 0.0;
+
+        state.dataRootFinder->HVACSystemRootFinding.HVACSystemRootSolver = HVACSystemRootSolverBackup;
+        return T;
```

Comment:
In general, this is so much better than it was, great job there.  I am still quite surprised to find the sheer amount of SolveRoot calls inside this module which is just trying to calculate heat index.  Anyway, thanks for the cleanups.

Reply:
Thanks for reviewing. Glad I've been able to clean it out a bit. It does seem a pretty involved calculation process in their paper as well.